src/lib.rs

```diff
@@ -563,25 +579,24 @@ impl<M: Manager> Pool<M> {
             }
         }
 
-        internals.num_open += 1;
-        drop(internals);
-
         log::debug!("get conn with manager create");
         match self.0.manager.connect().await {
             Ok(c) => {
+                internals.num_open += 1;
+                drop(internals);
+
                 let conn = Conn {
                     raw: Some(c),
                     last_err: Mutex::new(None),
                     created_at: Instant::now(),
                     last_used_at: Instant::now(),
+                    last_checked_at: Instant::now(),
                     brand_new: true,
                 };
 
                 return Ok(conn);
             }
             Err(e) => {
-                let internals = self.0.internals.lock().await;
```

Comment:
Maybe we should not remove them, cause the `num_open` leak was fixed at line [585](https://github.com/importcjj/mobc/pull/24/files#diff-b4aea3e418ccdb71239b96952d9cddb6R585).

Reply:
What does that `maybe_open_new_connection` really do here? Will it open a new connection and increment the counter if successful?

Reply:
When an inner get request fails, a retry will be made, and I expect to have an existing connection available at that time.

Reply:
I'm adding these two lines here and now the `test_lazy_initialization_failure` never finishes...

Reply:
Ah, of course it was the Mutex...